nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,281 @@
+    bg_img = load_mni152_template() if bg_img == 'MNI 152 Template' else bg_img
+    
+    with open(html_template_path) as html_file_obj:
+        html_template_text = html_file_obj.read()
+    
+    # print(html_template_text)
+    report_template = string.Template(html_template_text)
+    contrasts_display_text = pd.DataFrame.from_dict(contrasts, orient='index'
+                                                    ).to_html(border=0,
+                                                              header=False,
+                                                              )
+    pd.set_option('display.max_colwidth', 50)
+    model_attributes_html = make_model_attributes_html_table(model)
+    statistical_maps = make_statistical_maps(model, contrasts)
+    html_design_matrices = _report_design_matrices(model)
+    all_stat_map_cluster_table_pairs_html_code = (
+        _report_stat_maps_cluster_tables(statistical_maps,
+                                         threshold,
+                                         bg_img,
+                                         display_mode,
+                                         )
+    )
+    
+    report_values = {'title': 'Test Report',
+                     'model_attributes': model_attributes_html,
+                     'contrasts': contrasts_display_text,
+                     'design_matrices': html_design_matrices,
+                     'all_stat_map_cluster_table_pairs': all_stat_map_cluster_table_pairs_html_code,
+                     }
+    report_text = report_template.safe_substitute(**report_values)
+    print(report_text)
+    with open(output_path, 'w') as html_write_obj:
+        html_write_obj.write(report_text)
+
+
+def pretty_print_mapping_of_sequence(any_dict):
+    output_text = []
+    for key, value in any_dict.items():
+        formatted_value = pprint.pformat(value)
+        line_text = '{} : {}'.format(key, formatted_value)
+        output_text.append(line_text)
+    return '\n'.join(output_text)
+
+
+def make_model_attributes_html_table(model):
+    selected_model_attributes = {
+        attr_name: attr_val
+        for attr_name, attr_val in model.__dict__.items()
+        if not attr_name.endswith('_')
+        }
+    model_attributes_table = pd.DataFrame.from_dict(selected_model_attributes,
+                                                    orient='index',
+                                                    )
+    model_attributes_table = model_attributes_table.to_html(header=False,
+                                                            sparsify=False,
+                                                            )
+    return model_attributes_table
+
+
+def make_statistical_maps(model, contrasts):
+    """ Given a first model and contrasts, return the corresponding z-maps"""
+    statistical_maps = {}
+    for contrast_id, contrast_val in contrasts.items():
+        statistical_maps[contrast_id] = model.compute_contrast(
+                contrast_val)
```

Comment:
That is easy to do, but then why not just use the function externally and pass in the stats map, instead of re-exposing all the parameters and complicating it.

Reply:
see @bthirion 's comment: https://github.com/nistats/nistats/pull/352#discussion_r290927744
I guess you can think of the reporting function as an alternative to `compute_contrast` that presents its results in a different way, for human consumption rather machine consumption

Reply:
Hmm that's an interesting perspective. Might be useful.
Lets revisit this once we have a basic prototype ready.